### PR TITLE
DS-3179 Collection admin, edit item no authorization to remove file (updated PR)

### PR DESCRIPTION
**Pull request by PhilipVis**
https://jira.duraspace.org/browse/DS-3179

I made this pull request based on Kevin Van de Velde's pull request https://github.com/DSpace/DSpace/pull/1382#issuecomment-221902015 and on the changes proposed in the comments on this pull request.

What I did: 
- started from Kevin's pull request
- merged latest DSpace master into it 
- implemented the changes described by tdonohue 
- fixed any remaining bugs I encountered. Mainly making sure the bond between parent-child is severed after the delete instead of before, to make sure the authorization can be verified during the delete. 

I was able to successfully delete a bistream, bundle, item, collection and sub-community as community/collection administrator.

Please test these changes, because I might not have tested all cases.

### CODE REVIEW

**Comment by tdonohue** [comment]
+1, I've tested the following today, and it all seems to *work*!

Logged in as a Community Administrator, I can now:
- [x] Delete a sub-community
- [x] Delete a sub-collection
- [x] Delete an item under a sub-collection
- [x] Delete a single bitstream of such an item

Logged in as a Collection Administrator, I can now:
- [x] Delete an item under that collection
- [x] Delete a single bitstream of such an item

Prior to this PR, NONE of the following deletions succeed on `master` (in either UI). So, this PR seems to resolve the issues.  Thanks @PhilipVis !

(As a sidenote, it'd be nice to take this opportunity to create some integration tests that also validate the behavior which I manually tested above. They all seem like actions that could be auto-performed via integration tests.  But, this PR seems to fix the issue, and it's much better than the state of current master)

**Comment by tdonohue** [comment]
I've just created a companion PR to this one: #1447   It adds new Integration Tests which prove that this PR fixes the delete permissions of Community / Collection Administrators.

I'd highly recommend we merge this PR and #1447 as soon as possible.

File: dspace-api/src/main/java/org/dspace/content/BitstreamServiceImpl.java [hunk 1]
```diff
@@ -248,7 +248,12 @@ public void delete(Context context, Bitstream bitstream) throws SQLException, Au
         context.addEvent(new Event(Event.DELETE, Constants.BITSTREAM, bitstream.getID(),
                 String.valueOf(bitstream.getSequenceID()), getIdentifiers(context, bitstream)));
 
-        bitstream.getBundles().clear();
+        //Remove our bitstream from all our bundles
+        final List<Bundle> bundles = bitstream.getBundles();
+        for (Bundle bundle : bundles) {
+            bundle.getBitstreams().remove(bitstream);
+        }
+        //Remove all bundles from the bitstream object, clearing the connection in 2 ways
```
Comment (mwoodiupui) [inline dspace-api/src/main/java/org/dspace/content/BitstreamServiceImpl.java]: This comment is in a confusing place.  Shouldn't it move down to line 266?

**Comment by hardyoyo** [comment]
I've merged #1447 and #1443 as per tdonohue's instructions, and rebased the whole thing on current master, running mvn test -Dmaven.test.skip=false -DskipITs=false and all tests pass. I'd say this looks pretty good. +1

**Comment by tdonohue** [comment]
Already at +2, and approved in backlog hour today.  Merging!  Thanks @PhilipVis!

**Pull request merged**